### PR TITLE
423 test all product families

### PR DESCRIPTION
**Pull request by evrose54**
# Reviewer Checklist

* [x] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
* [x] NO REQUIRED ***unit tests*** (explain why not required)
* [x]  NO REQUIRED ***integration tests*** (explain why not required)
* [x] Required ***documentation*** added for new/modified functionality
* [x] Required ***release notes*** added for new/modified functionality
* [x] Required ***updates to other repos*** complete

# Related Issues
fixes NRLMMD-GEOIPS/geoips#423
fixes NRLMMD-GEOIPS/geoips#428

# Testing Instructions
Run geoips/tests/integration_tests/full_test.sh to check for sure. 

NOTE: I am having trouble installing the recenter_tc repo (akima86 build errors), so wait merging this until I can get that installed. Or if someone wants to, feel free to test this branch locally and make sure that is working.

# Summary

Create products for each possible product family handled by single_source procflow
----------------------------------------------------------------------------------

*From GEOIPS#423: 2024-1-17, Create products for each possible product family*

This update adds a new product ``all_product_families.yaml`` to the products interface.
This file provides every possible product family to test against ``single_source.py``
procflow plugin, as we need to understand which families work, and the process these
families follow throughout that code in order to implement a new procflow down the road,
called ``order_based.py`` procflow. By understanding the paths that these families take,
we set ourselves up for success in implementing that new procflow.

Files Added:

    - added: geoips/plugins/yaml/products/all_product_families.yaml
    - added: tests/scripts/infrared_all_product_families.sh

Note: See Bux Fix for duplicate source_names in product registry *GEOIPS#428*, for more
information on bug fixes made on this branch as well.

Fix product sub-plugins duplicate source name error
---------------------------------------------------

*From GEOIPS#428: 2024-2-6, source_name duplicate bug*

Currently, when the plugin registry is built, the key names associated with the
sub-products contained in a yaml file (for instance lets say smap-spd.yaml), are
gathered via the sub-products source name rather than the name of the product plugin
they reside in. In this case, all product plugins under the key name smap-spd, are
placed in that location because of their source name, not the attribute name, found
at the top level of smap-spd.yaml. This could cause issues with other products, as
found in issue *GEOIPS#423: 2024-1-17, create products for each possible product family*

This updates fixes this bug by changing the naming convention of product plugins stored
within the plugin registry. Instead of products -- <source_name> -- sub_product_name,
the keys are now formatted products -- <source_name>.<sub_product_name>, and duplicates
are checked on that level.

Files modified (excluding files which were modified for black formatting):

    -  modified: geoips/create_plugin_registries.py
    -  modified: geoips/interfaces/base.py
    -  modified: geoips/interfaces/yaml_based/products.py
    -  modified: geoips/dev/output_config.py
    -  modified: geoips/plugins/modules/filename_formatters/tc_file_naming.py
    -  modified: geoips/plugins/modules/output_formatters/imagery_annotated.py
    -  modified: geoips/plugins/modules/output_formatters/metadata_tc.py
    -  modified: geoips/plugins/modules/procflows/config_based.py
    -  modified: geoips/plugins/modules/procflows/single_source.py

### CODE REVIEW

**Comment by evrose54** [comment]
Don't merge until this can be tested on recenter_tc, and until v1.12.0-release has been merged.

File: geoips/xarray_utils/data.py [hunk 1]
```diff
@@ -788,9 +788,9 @@ def sector_xarrays(
                 sect_xarray.attrs["start_datetime"] = covg_xarray.start_datetime
                 sect_xarray.attrs["end_datetime"] = covg_xarray.end_datetime
 
-        sect_xarray.attrs["area_definition"] = (
```
Comment (evrose54) [inline geoips/xarray_utils/data.py:791]: Black formatting changes.

File: geoips/plugins/modules/readers/viirs_netcdf.py [hunk 2]
```diff
@@ -617,9 +617,9 @@ def call(fnames, metadata_only=False, chans=None, area_def=None, self_register=F
                         nparr.mask,
                     )
                     for attrname in ncvar.ncattrs():
-                        xarrays[geo_target_data_type][xvarname].attrs[attrname] = (
```
Comment (evrose54) [inline geoips/plugins/modules/readers/viirs_netcdf.py:620]: black formatting changes.

File: geoips/plugins/modules/readers/atms_hdf5.py [hunk 3]
```diff
@@ -323,9 +323,9 @@ def call(fnames, metadata_only=False, chans=None, area_def=None, self_register=F
         # MTIFs need to be "prettier" for PMW products, so 2km resolution for
         # final image
         xarray_atms.attrs["sample_distance_km"] = 2
-        xarray_atms.attrs["interpolation_radius_of_influence"] = (
```
Comment (evrose54) [inline geoips/plugins/modules/readers/atms_hdf5.py:326]: black formatting changes.

File: geoips/plugins/modules/procflows/single_source.py [hunk 4]
```diff
@@ -1703,9 +1701,9 @@ def call(fnames, command_line_args=None):
             # These will be added to the alg_xarray
             # new_attrs['area_definition'] = area_def
             if "adjustment_id" in area_def.sector_info:
-                new_attrs["filename_extra_fields"]["adjustment_id"] = (
```
Comment (evrose54) [inline geoips/plugins/modules/procflows/single_source.py:1706]: black formatting changes.

File: geoips/plugins/modules/procflows/single_source.py [hunk 5]
```diff
@@ -232,9 +232,9 @@ def combine_filename_extra_fields(source_xarray, dest_xarray):
         for field in source_xarray.filename_extra_fields:
             if "filename_extra_fields" not in dest_xarray.attrs:
                 dest_xarray.attrs["filename_extra_fields"] = {}
-            dest_xarray.attrs["filename_extra_fields"][field] = (
```
Comment (evrose54) [inline geoips/plugins/modules/procflows/single_source.py:235]: black formatting changes.

**Comment by jsolbrig** [comment]
@evrose54 I converted this to a draft until we have time to move the new products to the order_based procflow repo. The other changes here look good and should be merged.

File: docs/source/releases/v1_12_2a0.rst [hunk 6]
```diff
@@ -13,20 +13,84 @@
...
+
 * Bug fixes
 
   * Ensure that log.interactive works from plugins when imported independently.
   * Make add_log_level a callable class to fix AttributeErrors.
+  * Fix product source_name duplicate entries bug
+
+* Testing Updates
+
+  * Added all_product_families.yaml to test all available product families
```
Comment (mindyls) [inline docs/source/releases/v1_12_2a0.rst:25]: Adding some documentation on the product families as well would be good - even if it is just pointing to the test scripts as examples.

**Comment by jsolbrig** [comment]
Closed because the functionality from this has moved into other PRs.

**Pull request closed without merging**